File: LeetCode-Java/Palindrome Number.java
// this solution is right, but mostly out of luck
// negative cases?
// overflow cases? I guess it's right, however it's not guarenteed.
public class Solution {
    public boolean isPalindrome(int x) {
        int newX = 0;
        int y = x;
        while(y > 0)
        {
            newX = newX * 10 + y % 10;
            y = y / 10;
        }
        return newX == x;
    }
}

File: Leetcode-cpp/palindrome_number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0) return false;
        int mod = 1;
        while (x / 10 >= mod) mod *= 10;
        while (x > 0) {
            if (x % 10 != x / mod) return false;
            x = (x % mod) / 10;
            mod /= 100;
        }
        return true;
    }
};